### PR TITLE
Jordan Wigner tranform from spins to fermions

### PR DESCRIPTION
**Pull request by rreiner-hqs**

### CODE REVIEW

File: struqture/src/fermions/fermionic_noise_system.rs [hunk 1]
```diff
@@ -1,8 +1,4 @@
-// Copyright © 2021-2022 HQS Quantum Simulations GmbH. All Rights Reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
-// in compliance with the License. You may obtain a copy of the License at
-//
+// Copyright © 2021-2022 HQS Quantum Simulations GmbH. All Rights Reserved. // // Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except // in compliance with the License. You may obtain a copy of the License at //
```
Comment (kbarkhqs) [inline struqture/src/fermions/fermionic_noise_system.rs:1]: I think there's been a mistake here

File: struqture/src/fermions/fermionic_noise_system.rs [hunk 2]
```diff
@@ -1,7 +1,6 @@
 // Copyright © 2021-2022 HQS Quantum Simulations GmbH. All Rights Reserved.
 //
-// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
-// in compliance with the License. You may obtain a copy of the License at
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except // in compliance with the License. You may obtain a copy of the License at
```
Comment (kbarkhqs) [inline struqture/src/fermions/fermionic_noise_system.rs:3]: Typo here, should be on the next line

File: struqture/src/spins/spin_hamiltonian_system.rs [hunk 3]
```diff
@@ -259,7 +261,7 @@ impl SpinHamiltonianSystem {
     /// # Returns
     ///
     /// * `&SpinHamiltonian` - The SpinHamiltonian of the SpinHamiltonianSystem.
-    pub fn operator(&self) -> &SpinHamiltonian {
+    pub fn hamiltonian(&self) -> &SpinHamiltonian {
```
Comment (kbarkhqs) [inline struqture/src/spins/spin_hamiltonian_system.rs:264]: We need to discuss this change with Nico, as we may need to keep both functions, but deprecate the `operator` method

File: struqture/src/spins/spin_noise_operator.rs [hunk 4]
```diff
@@ -163,7 +165,8 @@ impl<'a> OperateOnDensityMatrix<'a> for SpinLindbladNoiseOperator {
     ///
     /// * `Ok(Some(CalculatorComplex))` - The key existed, this is the value it had before it was set with the value input.
     /// * `Ok(None)` - The key did not exist, it has been set with its corresponding value.
-    /// * `Err(StruqtureError)` - The input contained identities, which are not allowed as Lindblad operators.
+    /// * `Err(StruqtureError::InvalidLindbladTerms)` - The input contained identities, which are not allowed as Lindblad operators.
+
```
Comment (kbarkhqs) [inline struqture/src/spins/spin_noise_operator.rs:169]: Unnecessary line here

**Review by kbarkhqs — CHANGES_REQUESTED** [review]
Should review the operator to hamiltonian naming. Also, there are no tests for the FermionLindbladNoiseOperator function `add_from_full_operators`

File: struqture/src/spins/decoherence_operator.rs [hunk 5]
```diff
@@ -517,6 +519,26 @@ impl From<SpinOperator> for DecoherenceOperator {
...
+    /// The convention used is that |0> represents an empty fermionic state (spin-orbital),
+    /// and |1> represents an occupied fermionic state.
+    ///
+    /// # Returns
+    ///
+    /// `FermionOperator` - The fermionic operator that results from the transformation.
+    fn jordan_wigner(&self) -> Self::Output {
+        let mut out = FermionOperator::new();
+        for dp in self.keys() {
+            out = out + dp.jordan_wigner() * self.get(dp);
```
Comment (nfwvogt) [inline struqture/src/spins/decoherence_operator.rs:536]: Minor nitpick: we could directly use iter here
```Rust
for (dp, value) in self.iter(){
}
```

File: struqture/src/spins/decoherence_product.rs [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2021-2022 HQS Quantum Simulations GmbH. All Rights Reserved.
+// Copyright © 2020-2022 HQS Quantum Simulations GmbH. All Rights Reserved.
```
Comment (nfwvogt) [inline struqture/src/spins/decoherence_product.rs:1]: 2021 was correct in that file as far as I remember.

File: struqture/src/spins/spin_hamiltonian_system.rs (hunk 3 above)
Comment (nfwvogt) [reply struqture/src/spins/spin_hamiltonian_system.rs:264]: Yes we should add hamiltonian but keep operator with a deprecation warning.

**Review by nfwvogt — APPROVED** [review]

**Pull request merged**